src/js/subtraction/querys.ts

```diff
@@ -15,6 +23,23 @@ export const subtractionQueryKeys = {
     shortlist: () => ["subtraction", "list", "short"] as const,
 };
 
+/**
+ * Initializes a mutator for creating a subtraction
+ *
+ * @returns A mutator for creating a subtraction
+ */
+export function useCreateSubtraction() {
+    const history = useHistory();
+    return useMutation<Subtraction, unknown, { name: string; nickname: string; uploadId: string }>(
+        ({ name, nickname, uploadId }) => createSubtraction(name, nickname, uploadId),
+        {
+            onSuccess: () => {
+                history.push("/subtractions");
+            },
```

Comment:
You can and should specify navigation in the component instead of with the mutation.  You can specify `onSuccess` when you call `mutation.mutate` in the the component.

check out this post: https://tkdodo.eu/blog/mastering-mutations-in-react-query Covers the difference between specifying `onSuccess` on the `useMutation` vs `mutation.mutate` level